✅ Fix flakey integration test

The integration test is currently flakey, probably due to some faulty cleanup of the test environment.
```
✖  test/integration/k8s (6.83s)

=== Failed
=== FAIL: test/integration/k8s  (0.00s)
PASS
panic: Log in goroutine after TestIntegrationCapsuleReconcilerNginx has completed: "level"=0 "msg"="Stopping and waiting for non leader election runnables"


goroutine 389 [running]:
testing.(*common).logDepth(0xc000281520, {0xc0004e5e50, 0x49}, 0x3)
	/opt/hostedtoolcache/go/1.21.0/x64/src/testing/testing.go:1022 +0x4c5
testing.(*common).log(...)
	/opt/hostedtoolcache/go/1.21.0/x64/src/testing/testing.go:1004
testing.(*common).Log(0xc000281520, {0xc000395240?, 0x167a440?, 0xc000281501?})
	/opt/hostedtoolcache/go/1.21.0/x64/src/testing/testing.go:1045 +0x4a
github.com/go-logr/logr/testr.logInfo({0x1bb68a8, 0xc000281520}, 0xc000007d40?, 0xc000007d40?, {0x19917a3?, 0xc00006bd58?}, {0x0?, 0xc000007d40?, 0xc00006bd78?})
	/home/runner/go/pkg/mod/github.com/go-logr/logr@v1.2.4/testr/testr.go:101 +0x113
github.com/go-logr/logr/testr.testloggerInterface.Info({{0x0, {0x1bace18, 0x0}, {0x0, 0x0, 0x0}, {0x0, 0x0}, 0x1, 0xc0003b02d0}, ...}, ...)
	/home/runner/go/pkg/mod/github.com/go-logr/logr@v1.2.4/testr/testr.go:148 +0x106
sigs.k8s.io/controller-runtime/pkg/log.(*delegatingLogSink).Info(0xc000107800, 0x1bcc1f8?, {0x19917a3, 0x36}, {0x0, 0x0, 0x0})
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.1/pkg/log/deleg.go:127 +0xdd
github.com/go-logr/logr.Logger.Info({{0x1bcc1f8?, 0xc000107800?}, 0x0?}, {0x19917a3, 0x36}, {0x0, 0x0, 0x0})
	/home/runner/go/pkg/mod/github.com/go-logr/logr@v1.2.4/logr.go:278 +0xdc
sigs.k8s.io/controller-runtime/pkg/manager.(*controllerManager).engageStopProcedure.func3()
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.1/pkg/manager/internal.go:516 +0x53
created by sigs.k8s.io/controller-runtime/pkg/manager.(*controllerManager).engageStopProcedure in goroutine 167
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.1/pkg/manager/internal.go:514 +0x258
FAIL	github.com/rigdev/rig/test/integration/k8s	6.828s

DONE 78 tests, 1 failure in 10.035s
make: *** [Makefile:78: test-all] Error 1
Error: Process completed with exit code 2.
```

--- a/test/integration/k8s/capsule_controller_test.go
+++ b/test/integration/k8s/capsule_controller_test.go
@@ -14,8 +14,8 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 
-	"github.com/rigdev/rig/pkg/controller"
 	"github.com/rigdev/rig/pkg/api/v1alpha1"
+	"github.com/rigdev/rig/pkg/controller"
 
 	//+kubebuilder:scaffold:imports
 
@@ -29,7 +29,7 @@ func TestIntegrationCapsuleReconcilerNginx(t *testing.T) {
 	t.Parallel()
 
 	env := setupTest(t, options{runManager: true})
-	defer env.cancel()
+	defer env.stop()
 	k8sClient := env.k8sClient
 
 	ctx := context.Background()

--- a/test/integration/k8s/setup_test.go
+++ b/test/integration/k8s/setup_test.go
@@ -5,6 +5,7 @@ import (
 	"fmt"
 	"path/filepath"
 	"runtime"
+	"sync"
 	"testing"
 	"time"
 
@@ -35,6 +36,12 @@ type options struct {
 type env struct {
 	k8sClient client.Client
 	cancel    context.CancelFunc
+	wg        sync.WaitGroup
+}
+
+func (e *env) stop() {
+	e.cancel()
+	e.wg.Wait()
 }
 
 func setupTest(t *testing.T, opts options) *env {
@@ -63,6 +70,21 @@ func setupTest(t *testing.T, opts options) *env {
 
 	ctx, cancel := context.WithCancel(context.Background())
 
+	e := &env{
+		k8sClient: k8sClient,
+		cancel:    cancel,
+		wg:        sync.WaitGroup{},
+	}
+
+	e.wg.Add(1)
+	go func() {
+		defer e.wg.Done()
+		<-ctx.Done()
+		if err := testEnv.Stop(); err != nil {
+			fmt.Printf("could not stop envtest: %s\n", err)
+		}
+	}()
+
 	if opts.runManager {
 		manager, err := ctrl.NewManager(cfg, ctrl.Options{
 			Scheme:  scheme,
@@ -83,19 +105,12 @@ func setupTest(t *testing.T, opts options) *env {
 
 		assert.NoError(t, capsuleReconciler.SetupWithManager(manager))
 
+		e.wg.Add(1)
 		go func() {
+			defer e.wg.Done()
 			assert.NoError(t, manager.Start(ctx))
 		}()
 	}
 
-	go func() {
-		<-ctx.Done()
-		testEnv.ControlPlane.Etcd.StopTimeout = time.Second * 30
-		assert.NoError(t, testEnv.Stop())
-	}()
-
-	return &env{
-		cancel:    cancel,
-		k8sClient: k8sClient,
-	}
+	return e
 }

